src/catalog/src/information_schema/columns.rs

```diff
@@ -236,7 +332,94 @@ impl InformationSchemaColumnsBuilder {
         self.schema_names.push(Some(schema_name));
         self.table_names.push(Some(table_name));
         self.column_names.push(Some(&column_schema.name));
-        self.data_types.push(Some(data_type));
+
+        self.ordinal_positions.push(Some((index + 1) as i64));
+
+        if column_schema.data_type.is_string() {
+            self.character_maximum_lengths.push(Some(MAX_STRING_LENGTH));
+            self.character_octet_lengths.push(Some(MAX_STRING_LENGTH));
+            self.numeric_precisions.push(None);
+            self.numeric_scales.push(None);
+            self.datetime_precisions.push(None);
+            self.character_set_names.push(Some(UTF8_CHARSET_NAME));
+            self.collation_names.push(Some(UTF8_COLLATE_NAME));
+        } else if column_schema.data_type.is_numeric() || column_schema.data_type.is_decimal() {
+            self.character_maximum_lengths.push(None);
+            self.character_octet_lengths.push(None);
+
+            match &column_schema.data_type {
+                ConcreteDataType::Int8(_) | ConcreteDataType::UInt8(_) => {
+                    self.numeric_precisions.push(Some(3));
+                    self.numeric_scales.push(Some(0));
+                }
+                ConcreteDataType::Int16(_) | ConcreteDataType::UInt16(_) => {
+                    self.numeric_precisions.push(Some(5));
+                    self.numeric_scales.push(Some(0));
+                }
+                ConcreteDataType::Int32(_) | ConcreteDataType::UInt32(_) => {
+                    self.numeric_precisions.push(Some(10));
+                    self.numeric_scales.push(Some(0));
+                }
+                ConcreteDataType::Int64(_) => {
+                    self.numeric_precisions.push(Some(19));
+                    self.numeric_scales.push(Some(0));
+                }
+                ConcreteDataType::UInt64(_) => {
+                    self.numeric_precisions.push(Some(20));
+                    self.numeric_scales.push(Some(0));
+                }
+                ConcreteDataType::Float32(_) => {
+                    self.numeric_precisions.push(Some(12));
+                    self.numeric_scales.push(None);
+                }
+                ConcreteDataType::Float64(_) => {
+                    self.numeric_precisions.push(Some(22));
+                    self.numeric_scales.push(None);
+                }
+                ConcreteDataType::Decimal128(decimal_type) => {
+                    self.numeric_precisions
+                        .push(Some(decimal_type.precision() as i64));
+                    self.numeric_scales.push(Some(decimal_type.scale() as i64));
```

Comment:
how about moving `numeric_precisions` and `numeric_scales` to `ConcreteDataType`?

Reply:
Looks good to me, I'll change it.